### ISSUE TITLE
Horizontal centering

### ISSUE THREAD

**Issue by fr500** [vurtun/nuklear#676]
I can't seem to be able to center content on screen.

![image](https://user-images.githubusercontent.com/1721040/39090491-7c369bc2-45a6-11e8-8d8c-14b13f07eb00.png)

I have the following code:

```C
int main(void)
{
   setup();
   struct nk_color background;
   theme_idx = THEME_BLUE_GREY_NIGHT;
   old_theme_idx = theme_idx;
   
   set_style(ctx, theme_idx, 0, true);

   while (!glfwWindowShouldClose(win))
   {
      nk_glfw3_new_frame();
      file_picker(ctx, width, height, NULL, NULL, NULL);
      -----irrelevant stuff------
```

`file_picker` looks like this:

```
bool file_picker(struct nk_context *ctx, unsigned width, unsigned height, 
   char *ret, void (*data_cb), void (*action_cb))
{
   size_t item_size, row_padding;

   float win_padding = width * 0.05;

   if (nk_begin(ctx, "File Picker", nk_rect(0, 0, width, height),
      NK_WINDOW_NO_SCROLLBAR))
   {
      /* padding */
      nk_layout_row_static(ctx, win_padding, 1, 1);

      /* header row */
      set_font_index(ctx, fonts_tenfoot_bold, 4);
      item_size = fonts_tenfoot_bold[4].size + 20;
      row_padding = 20;

      nk_layout_row_begin(ctx, NK_STATIC, item_size + row_padding, 3);
      {
         /* padding */
         nk_layout_row_push(ctx, win_padding);
         nk_label(ctx, "", NK_TEXT_LEFT);

         /* header */
         nk_layout_row_push(ctx, width * 0.90);
         nk_tenfoot_header(ctx, "Header Area", ICON_FA_FOLDER, item_size, row_padding);

         /* padding */
         nk_layout_row_push(ctx, win_padding);
         nk_label(ctx, "", NK_TEXT_LEFT);
      }
      nk_layout_row_end(ctx);
      -----irrelevant stuff------
```

And `nk_tenfoot_header` looks like this:

```
void nk_tenfoot_header(struct nk_context *ctx, const char* label, const char* icon, size_t size, size_t pad)
{
   nk_group_begin(ctx, label, NK_WINDOW_BORDER);
   {
      nk_layout_row_begin(ctx, NK_STATIC, size, 3);
      {
         nk_layout_row_push(ctx, size);
         nk_button_label(ctx, icon);
         nk_layout_row_push(ctx, pad);
         nk_label(ctx, "", NK_TEXT_LEFT);
         nk_layout_row_push(ctx, width * 0.90 - 2 * size);
         nk_label(ctx, label, NK_TEXT_LEFT);
      }
      nk_layout_row_end(ctx);
   }
   nk_group_end(ctx);
}
```

The header and content areas are obviously not centered, I tried playing with padding, scrollbar size, etc but I haven't figured it properly

Maybe there is something wrong in my usage of the library?

Basically I'm drawing like this:

PADDING COL | HEADER COL | PADDING COL

Maybe I should just use nk_layout_row_static?
Is that the correct way to position widgets on screen?

Similar to this?
https://github.com/vurtun/nuklear/issues/515#issuecomment-330033323

Adding padding columns is probably cheap, lazy, I just want to know if the row_static approach would be better

**Comment by fr500** [comment]
Hey, thanks for responding!

It basically adds an empty row on top. When I added the extra spacing it added... two rows I think.

Current code:
![image](https://user-images.githubusercontent.com/1721040/39096146-8ce8ee50-4611-11e8-89a7-f1ba64a25b05.png)

Without `nk_layout_row_static`
![image](https://user-images.githubusercontent.com/1721040/39096154-a25954b4-4611-11e8-83ee-f3c490071af9.png)

Extra spacing:
![image](https://user-images.githubusercontent.com/1721040/39096157-b948f1ac-4611-11e8-85a6-e9d42d98b8a8.png)

In this case vertical positioning is not an issue, actually the code that's being problematic for me is the code that makes the columns I guess

For instance the header area is this code:

```C
      nk_layout_row_begin(ctx, NK_STATIC, item_size + row_padding, 3);
      {
         /* padding */
         nk_layout_row_push(ctx, win_padding);
         nk_label(ctx, "", NK_TEXT_LEFT);

         /* header */
         nk_layout_row_push(ctx, width * 0.90);
         nk_tenfoot_header(ctx, "Header Area", ICON_FA_FOLDER, item_size, row_padding);

         /* padding */
         nk_layout_row_push(ctx, win_padding);
         nk_label(ctx, "", NK_TEXT_LEFT);
      }
      nk_layout_row_end(ctx);
```

What it does is add two empty columns size 5% of the window  (`float win_padding = width * 0.05;`), and the important one sized 90% of the window.

My assumption is that it's actually drawing them the same exact size but for reasons I don't understand the left padding column ends up being too big.

Regarding this:

```
idenote:
instead of nk_label(ctx, "", NK_TEXT_LEFT); you can use nk_spacing to get an empty widget slot.
```

I tried actually, but it would crash on me if I make it like that for the padding AFTER, the header item

```

```
This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.
Assertion failed!

Program: E:\Tools\msys64\home\Andres\repositories\simpleton\out\nuklear-tk.exe
File: ./deps/nuklear/nuklear.h, Line 18584
```
         /* padding */
         nk_layout_row_push(ctx, win_padding);
         nk_spacing(ctx, 1);

         /* header */
         nk_layout_row_push(ctx, width * 0.90);
         nk_tenfoot_header(ctx, "Header Area", ICON_FA_FOLDER, item_size, row_padding);

         /* padding */
         nk_layout_row_push(ctx, win_padding);
         nk_spacing(ctx, 1); ----->crashes
```

https://git.polygoncorp.com/radius/simpleton/blob/master/src/filemon.c

That's the full source for reference.

**Comment by vurtun** [comment]
> Now the question is, what does win->padding and win->spacing control? the spacing and padding between adjacent windows? or the item padding and spacing **inside ** a window?

> he item padding and spacing **inside ** a window

Yes. `nk_style_window` has padding and spacing definition for each panel content type.
